### PR TITLE
Brainstorming for #95

### PR DESCRIPTION
**Pull request by pllim**
As discussed in #95, here are some ideas for the `Mosaic` plugin. Again, please ignore my Emacs automatically removing trailing whitespaces. I could not effectively catch the `MemoryError`, but if we calculate FOV dynamically (or even just set the default smaller), we can avoid the error from happening in the first place.

* I added error class name to be displayed in `Control.py` (Line 420). I find this useful.
* I unchecked "Drop New" by default. Intuitively for me, when I drop new file, I want to add them to the existing mosaic, not overwrite it.
* I added `_calc_fov()` method in `Mosaic.py` to roughly estimate the FOV needed to contain all the images. However, this does not work if user created the mosaic from one image, and then drop in a different image separately. It works if user CNTRL+click on all the images and drop them in at once. Like you said, there can be a separate method that stores all the RA and DEC info and recalculate FOV when user drops in new file separately, and use that to create bigger mosaic and paste the existing mosaic in, but I can't implement that without butchering the existing codes. Maybe you know better where to stick that kind of logic in?
* I disabled creation of new mosaic if `dist > fov_deg` because it is not a behavior I want. But I guess we don't have to disable this if we can calculate `fov_deg` such that it is always bigger than `dist`.
* You can ignore the changes in `dp.py`. I inserted memory check there but it didn't stop `MemoryError` from happening.

### CODE REVIEW

**Comment by ejeschke** [comment]
Hi @pllim, can you check out commit bc0f9600ffacd228fc1c16c66effb3620cdd6e1f before we go further with this PR?  This adds support for dynamically expanding the mosaic when new tiles are dropped.  See `examples/configs/plugin_Mosaic.cfg` for all the possible settings.  Some of these need to be exposed so they can be changed on the fly in the GUI--this is TBD.

With this behavior, I'm wondering if we need to do an initial pass to calculate the fov.

One other issue with your PR as it stands is that it assumes the reference pixel for the WCS is in the center of the image tiles.  This is true for your sample images, but not required in practice, I think (in fact the reference pixel need not be contained in the data part of the file).  So some typical mosaics that I tried with Subaru instrument files failed.  If we keep the fov calculation (which we maybe could as an option), we'll need to account for this.

**Comment by pllim** [comment]
@ejeschke , I like your changes. It does not look like we need to pre-calculate FOV anymore. So, you can probably ignore this PR, because I opened another PR for the one change that I still wish to see for the traceback.

Regarding your changes, I notice that when I set my preference for `Mosaic` to create new thumbnails, I see the new entries in `Contents` but not `Thumbs`. In addition, when I have "label images" turned on, a new label overwrites all the previous ones in all the mosaic entries I see in `Contents`, which does not make sense visually.

**Comment by ejeschke** [comment]
I see what you are saying about the labels.  It only works if the image is not resized. That's going to require some more drastic workaround, like clearing and redrawing all the labels if the image changes.  Until that happens, I've added a test to only draw the labels if allow_expand==False.

Regarding the creation of the Thumb for a mosaic, I think the problem is that there is no path associated with the mosaic.  Thumbs uses a file's path as part of a key to identify the thumb (since there can be different files with the same name and also because if a thumbnail is cached).  I think Contents doesn't actually check for a path, since it sorts files by channel.

Then there is also the problem that if the data drops out of the `Datasrc` then clicking on the thumbnail should reconstruct the mosaic from the tiles.  We actually have this capability in the subclassed plugin we use here, but it only creates Contents entries, not Thumbs entries.  The new ability to keep dropping new tiles on the mosaic could also complicate this.

In short, this is a bit sticky problem and requires some consideration.

**Comment by pllim** [comment]
It is fine for me if `Thumbs` does not display mosaic thumbnails, as long as it is documented clearly. I like that I can still access it from `Contents`.

Currently, this entry in `plugin_Mosaic.cfg` is very misleading:
```
# Allow mosaic images to create thumbnail entries
make_thumbs = False
```

**Comment by pllim** [comment]
In addition, the mosaic names in `Contents` are not very descriptive. Perhaps a new field in `Mosaic` plugin to allow user to overwrite mosaic name (or at least, the `dp` prefix) when user starts a new one?
